chronos/asyncmacro2.nim

```diff
@@ -272,7 +272,19 @@ template await*[T](f: Future[T]): auto =
       var chronosInternalTmpFuture {.inject.}: FutureBase
     chronosInternalTmpFuture = f
     chronosInternalRetFuture.child = chronosInternalTmpFuture
+
+    # This "yield" is meant for a closure iterator in the caller.
     yield chronosInternalTmpFuture
+
+    # By the time we get control back here, we're guaranteed that the Future we
+    # just yielded has been completed, through a very complicated mechanism in
```

Comment:
we're guaranteed that the Future just yielded has changed its state.

Reply:
Clarified.